server/modifiers.py

```diff
@@ -46,7 +47,19 @@ def output(self) -> list[str]:
         for method in list(methods):
             method(self)
 
-        return self.modified_contents
+        diff = DeepDiff(self.file_contents, self.modified_contents)
+        line_diffs = []
+
+        try:
+            for line_num, values in diff["values_changed"].items():
+                (num,) = list(filter(lambda x: x.isdigit(), re.split(r"(\d*)", line_num)))
+                new_value = values["new_value"]
+                line_diffs.append((int(num), new_value))
+        except KeyError:
+            # No values were changed
+            return [(0, "")]
```

Comment:
My only worry about this is when sending data back to the client to modify specific lines of code. If no values were changed, this will still return with the first line of code becoming an empty string. Is this acceptable or would it be better to do something else? Perhaps re-run the code again until there is no KeyError or check for this specific value when sending a request back to the client and abort the request if that is the case?

Reply:
Couldn't it just return an empty list?

Reply:
Yeah, that could work. I think then we would just need a check in place before looping over the output data.